cli/js/test_util.ts

```diff
@@ -137,42 +137,65 @@ function assertOps(fn: Deno.TestFunction): Deno.TestFunction {
 // the test has exactly the same contents as before the test.
 function assertResources(fn: Deno.TestFunction): Deno.TestFunction {
   return async function resourceSanitizer(): Promise<void> {
-    const preResources = Deno.resources();
+    const pre = Deno.resources();
     await fn();
-    const postResources = Deno.resources();
+    const post = Deno.resources();
     const msg = `Test case is leaking resources.
-Before: ${JSON.stringify(preResources, null, 2)}
-After: ${JSON.stringify(postResources, null, 2)}`;
-
-    assertEquals(preResources, postResources, msg);
+    Before: ${JSON.stringify(pre, null, 2)}
+    After: ${JSON.stringify(post, null, 2)}`;
+    assertEquals(pre, post, msg);
   };
 }
 
-export function testPerm(perms: TestPermissions, fn: Deno.TestFunction): void {
-  const normalizedPerms = normalizeTestPermissions(perms);
+interface UnitTestOptions {
+  skip?: boolean;
+  perms?: TestPermissions;
+}
 
-  registerPermCombination(normalizedPerms);
+export function unitTest(fn: Deno.TestFunction): void;
+export function unitTest(options: UnitTestOptions, fn: Deno.TestFunction): void;
+export function unitTest(
+  optionsOrFn: UnitTestOptions | Deno.TestFunction,
+  maybeFn?: Deno.TestFunction
+): void {
+  assert(optionsOrFn, "At least one argument is required");
+
+  let options: UnitTestOptions;
+  let name: string;
+  let fn: Deno.TestFunction;
+
+  if (typeof optionsOrFn === "function") {
+    options = {};
+    fn = optionsOrFn;
+    name = fn.name;
+    assert(name, "Missing test function name");
+  } else {
+    options = optionsOrFn;
+    assert(maybeFn, "Missing test function definition");
+    assert(
+      typeof maybeFn === "function",
+      "Second argument should be test function definition"
+    );
+    fn = maybeFn;
+    name = fn.name;
+    assert(name, "Missing test function name");
+  }
 
-  if (!permissionsMatch(processPerms, normalizedPerms)) {
+  if (options.skip) {
     return;
   }
```

Comment:
Not necessary for this patch, but I think in the future we should display a message for skipped tests. At least in Rust, seeing the "ignored" in the test output reminds us that we're ignoring them.

Reply:
I agree, I intend to integrate it into `runTests()` workflow (#4235, #4237)